### PR TITLE
Merge ocaml-rrdd-plugin into xcp-rrdd

### PR DESCRIPTION
**Pull request by lippirk**
The procedure followed is much the same as https://github.com/xapi-project/xcp-rrdd/pull/117:

 - git subtree add --prefix lib/plugin git@github.com:xapi-project/ocaml-rrdd-plugin master
 - tidy up lib/plugin by deleting files and moving files around, e.g. ocaml-plugin-tests are moved to test/plugin

### CODE REVIEW

File: rrdd-plugin.opam [hunk 1]
```diff
@@ -0,0 +1,29 @@
+opam-version: "2.0"
+maintainer: "xen-api@lists.xen.org"
+authors: "xen-api@lists.xen.org"
+homepage: "https://xapi-project.github.io/"
+bug-reports: "https://github.com/xapi-project/xcp-rrdd/issues"
+dev-repo: "git://github.com/xapi-project/xcp-rrdd.git"
+build: [["dune" "build" "-p" name "-j" jobs]]
```
Comment (psafont) [inline rrdd-plugin.opam:7]:
```suggestion
build: [
  ["dune" "build" "-p" name "-j" jobs]
  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
]
```

**Comment by lippirk** [comment]
I decided to get rid of the rrdp_dummy test - all it does is call `rrdp_dummy.exe --help`...

Perhaps we could remove the rrdp_dummy executable altogether; the argument against that is that we should keep it for documentation reasons. I'm not particularly fussed either way

**Review by psafont — APPROVED** [review]
The integration looks good, it's a shame there doesn't seem to be any tests :(

**Review by lindig — APPROVED** [review]

**Review by robhoes — APPROVED** [review]
Looks good to me.

@lippirk Could you squash the final fixup please?

**Pull request merged**